src/components/TokenEnterAmount.tsx

```diff
@@ -272,6 +272,26 @@ export function useEnterAmount(props: {
   }
 }
 
+interface Props {
+  token?: TokenBalance
+  inputValue: string
+  tokenAmount: string
+  localAmount: string
+  amountType: AmountEnteredIn
+  inputRef: React.MutableRefObject<RNTextInput | null>
+  inputStyle?: StyleProp<TextStyle>
+  autoFocus?: boolean
+  editable?: boolean
+  testID?: string
+  onInputChange(value: string): void
+  toggleAmountType?(): void
+  onOpenTokenPicker?(): void
+
+  /** Used in order to show available balance.
+   * @default token.balance  */
+  tokenBalance?: TokenBalance['balance']
```

Comment:
`processedAmounts` is currently still using `props.token.balance`, is this intentional? should we perhaps make the prop required? i feel like having a `tokenBalance` and `token.balance` is confusing because the two are so similar...what do you think about having the `EarnEnterAmount` modify the `token` instead? i.e. depending on what kind of earn flow, the `token` passed to the enter amount component changes. this might be a terrible idea :| but the benefit would be that we contain the token balance logic within the scope of where it's needed?

Reply:
Hmm, this makes sense! I'll try it out and if it doesn't complicate things a lot - will commit the changes.

Reply:
This worked and was updated!